Sources/PointFree/Account/UpdateProfile.swift

```diff
@@ -22,24 +24,43 @@ extension ProfileData: Decodable {
     let container = try decoder.container(keyedBy: CodingKeys.self)
 
     self.email = try container.decode(EmailAddress.self, forKey: .email)
-    self.name = try container.decodeIfPresent(String.self, forKey: .name)
     self.emailSettings = (try? container.decode([String: String].self, forKey: .emailSettings)) ?? [:]
+    self.extraInvoiceInfo = try? container.decode(String.self, forKey: .extraInvoiceInfo)
+    self.name = try container.decodeIfPresent(String.self, forKey: .name)
   }
 }
 
 func isValidEmail(_ email: EmailAddress) -> Bool {
   return email.rawValue.range(of: "^.+@.+$", options: .regularExpression) != nil
 }
 
+private func fetchStripeSubscription<A>(
```

Comment:
we have a middleware transformer that is similar to this except it was to produce an honest `Stripe.Subscription` and have an error otherwise, whereas for this screen I'm ok with not being able to find a stripe subscription. I tried to transform it in some way to be able to reuse, but couldn't figure it out.

Reply:
oh I think I know how this should be done, but don't really wanna spend too much more time on it.

the idea is that the `fetchStripeSubscription` middleware transformer in `Cancel.swift` should really just be middleware since it doesn't need to do any state changes. with that done I think it should then be possible to `map` over it and make it pluggable into what I'm doing here.

can look at this another time, just writing this for posterity.